src/document/HolonomicPathStore.ts

```diff
@@ -244,6 +255,16 @@ export const HolonomicPathStore = types
           const root = getRoot<IStateStore>(self);
           root.select(self.waypoints[0]);
         }
+
```

Comment:
Looks like your message cut out the logic. Are you talking about this logic:
```java
if (self.visibleWaypointsRange.length == 0) {
  this.setVisibleWaypoints([0, self.waypoints.length - 1]);
} else if (self.visibleWaypointsRange[1] == self.waypoints.length - 2) {
  this.setVisibleWaypoints([
    self.visibleWaypointsRange[0],
    self.waypoints.length - 1,
  ]);
}
```

If so, that just checks if you're currently focusing on the last waypoint and if so, when you add a new waypoint, to automatically add one to the end index of the focused waypoints so it focuses on the new waypoint as well. It also remedies the problem of `visibleWaypointsRange` starting out as an empty number array since if it detects that and you add a waypoint, it just sets it so that all the currently added waypoints are visible.

Honestly it might be better to focus the whole path if you add a new waypoint although that might feel a little jank if by adding a waypoint your whole focus is ruined.

Should I add a comment explaining this logic?

Reply:
You should definitely comment it. It does assume that waypoints added are always at the end of the list, which is true at the moment they're added, but in some cases (like pasting) they are immediately reordered.

Reply:
Hm fair I didn't think of that logic case. Didn't realize you could even paste stuff lol. I removed that logic and replaced it with a significantly simplified version. Will comment with an explanation later tonight.